t/t1028-mf-priority-projects.t

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+test_description='test validating and setting project names in priority plugin'
+
+. `dirname $0`/sharness.sh
+MULTI_FACTOR_PRIORITY=${FLUX_BUILD_DIR}/src/plugins/.libs/mf_priority.so
+SUBMIT_AS=${SHARNESS_TEST_SRCDIR}/scripts/submit_as.py
+DB_PATH=$(pwd)/FluxAccountingTest.db
+
+export TEST_UNDER_FLUX_NO_JOB_EXEC=y
+export TEST_UNDER_FLUX_SCHED_SIMPLE_MODE="limited=1"
+test_under_flux 1 job
+
+flux setattr log-stderr-level 1
+
+test_expect_success 'load multi-factor priority plugin' '
+	flux jobtap load -r .priority-default ${MULTI_FACTOR_PRIORITY}
+'
+
+test_expect_success 'check that mf_priority plugin is loaded' '
+	flux jobtap list | grep mf_priority
+'
+
+test_expect_success 'create flux-accounting DB' '
+	flux account -p $(pwd)/FluxAccountingTest.db create-db
+'
+
+test_expect_success 'start flux-accounting service' '
+	flux account-service -p ${DB_PATH} -t
+'
+
+test_expect_success 'add banks to the DB' '
+	flux account -p ${DB_PATH} add-bank root 1 &&
+	flux account -p ${DB_PATH} add-bank --parent-bank=root account1 1
+'
+
+test_expect_success 'add projects to the DB' '
+	flux account -p ${DB_PATH} add-project projectA &&
+	flux account -p ${DB_PATH} add-project projectB &&
+	flux account -p ${DB_PATH} add-project projectC
+'
+
+test_expect_success 'submit a job under a user before plugin gets updated' '
+	job0=$(flux python ${SUBMIT_AS} 1003 hostname) &&
+	test $(flux jobs -no {state} ${job0}) = PRIORITY
+'
+
+test_expect_success 'add user to flux-accounting DB and to plugin; job transitions to RUN' '
+	flux account -p ${DB_PATH} add-user --username=user1003 --userid=1003 --bank=account1 &&
+	flux account-priority-update -p ${DB_PATH} &&
+	test $(flux jobs -no {state} ${job0}) = RUN
```

Comment:
There's a slight race condition here (and actually above in the check for PRIORITY) because of the eventual consistency properties of `job-list`. It might be safer to check the eventlog and ensure the expected _event_ has been posted.

E.g. instead of checking that the job is in the PRIORITY state above, check that the last event in the eventlog is `depend`.

and instead of checking for the `RUN` state here, wait for an `alloc` event...